whisk/client.go

```diff
@@ -676,12 +676,10 @@ func IsResponseResultSuccess(data []byte) bool {
 	errResp := new(WhiskErrorResponse)
 	err := json.Unmarshal(data, &errResp)
 
-	if err != nil {
-		Debug(DbgWarn, "IsResponseResultSuccess: failed to parse response result: %v\n", err)
-		return false
-	}
-
 	if errResp.Response != nil {
+		if err != nil && !errResp.Response.Success { //failed to parse WhiskErrorResponse AND was not success
```

Comment:
@tysonnorris - i'm trying to think of a scenario in which errResponse.Response != nil and err != nil..  that is, if the bytes can't be parsed (i.e. not valid json) `err` will be set to some error (!= nil) and `errResp` will be left alone (pointer to struct having the `Response` field pointer set to nil).  but i'm guessing you had some response that drove this logic, so would you mind pasting that response string/bytes here to set me straight :smile:

Reply:
I don't have any example, but I agree the logic is wrong; maybe it should rather be:
```	if errResp.Response != nil {
		return errResp.Response.Success
	} else if err != nil { //failed to parse WhiskErrorResponse AND was not success
		Debug(DbgWarn, "IsResponseResultSuccess: failed to parse response result: %v\n", err)
	}```
WDYT?

Reply:
@dubee was just trying to avoid displaying the misleading warning message described in #89; however, the function contract was changed such that a `false` was returned when unable to parse the results.  previously, the function returned a `true` even when the results could not be parsed - the assumption being the http response was a successful 2xx and this function was only checking for a specific response body format that included a secondary success/failure status.

that said, your proposal looks good with the full function looking like
```
	errResp := new(WhiskErrorResponse)
	err := json.Unmarshal(data, &errResp)

	if errResp.Response != nil {
		return errResp.Response.Success
	} else if err != nil { //failed to parse WhiskErrorResponse
		Debug(DbgWarn, "IsResponseResultSuccess: failed to parse response result: %v\n", err)
	}

	return true
```